index.html

```diff
@@ -88,49 +96,224 @@
+
+    svg.append("g")         
+        .attr("class", "grid")
+        .call(make_y_axis()
+            .tickSize(-width, 0, 0)
+            .tickFormat("")
+        )
+    
     // HISTOGRAMS
 
-    // REGRESSION LINE
+    var formatCount = d3.format(",.0f");
+
+    //var margin = {top: 10, right: 30, bottom: 30, left: 30},
+        //width = 960 - margin.left - margin.right,
+        //height = 500 - margin.top - margin.bottom;
+
+    //Alex Answer: Yes, you do need to recreate the scale for the height of the rects in the histogram. Now, for each histogram, you are going to want a linear scale. This scale will have range and a domain, like other scales. 
+    //The domain should refer to your input data: What are the lowest and highest values that are important to the height of a histogram?
+    //The range should still be for the range of the output space, starting at 0, and going to the maximum height you would like the bars to going.
+    //Note that in the image Chris attached to the assignment, there is only one axis for each variable. So you don't have to recreate the axes.
+    
+    
+    xh = d3.scale.linear()
+        .rangeRound([0, width]);
+
+    /*
+    var bins = d3.layout.histogram()
+        .domain(xh.domain())
+        .thresholds(xh.ticks(4))
+        (data);
+    */
+
+    // Number of beans for X = 4
+    // x = min value of the data
+    // dx = distance between each bar 
+    // y = max value of the data 
+
+    //adding values ramdomly a bar is created. 
+    var bins = d3.layout.histogram()
+        .bins( [0, 0.3, 15] )(data)
+        //.thresholds(xh.ticks(4));
+        
+    yh = d3.scale.linear()
+        .domain([0, d3.max(bins, function(data) { return data.length; })])
+        .range([height, 0]);
+    
+   
+    var bar = svg.selectAll(".bar")
+        .data(bins)
+      .enter().append("g")
+        .attr("class", "bar")
+        .attr("transform", function(bin) {
+        //what is in x, dx, and y?  
+          console.log( "x: " + bin.x );
+          console.log( "dx: " + bin.dx );
+          console.log( "y: " + bin.y );
+
+          return "translate(" + xh(bin.dx) + "," + yh(bin.length) + ")"; 
+        });
+        //.attr("transform", function(bin) { return "translate(5," + yh(bin.length) + ")"; });
+
+    bar.append("rect")
+        .attr("x", 1)
+        //.attr("width", xh(bins[0].x1) - xh(bins[0].x0) - 1)
+        .attr("width", 10)
+        //.attr("height", function(bin) { return height - y(bin.length); });
+        .attr("height", 5);
+
+    bar.append("text")
+        .attr("dy", ".75em")
+        .attr("y", 0)
+        //.attr("x", (xh(bins[0].x1) - xh(bins[0].x0)) / 2)
+        .attr("x", 0)
+        .attr("text-anchor", "middle")
+        .text(function(d) { return formatCount(d.length); });
+
+    svg.append("g")
+        .attr("class", "axis axis--x")
+        .attr("transform", "translate(0," + height + ")")
+        //.call(d3.axis(xh).orient("bottom"));
+        .call(xAxis);
```

Comment:
A third X axis? (And actually, this would get created every time `update()` is called, so we'll just keep adding new axes to the chart. Remember that `setup()` is only called once, while `update()` is called repeatedly.)